QuizApp/App/Data sources/Quiz/QuizNetworkDataSource.swift

```diff
@@ -7,6 +7,8 @@ protocol QuizNetworkDataSourceProtocol {
 
     func fetchQuizes(for category: CategoryDataModel) async throws -> [QuizDataModel]
 
+    func startQuizSession(for id: Int) async throws -> [QuestionDataModel]
```

Comment:
I would consider renaming this to `getQuestions(for quizId: Int) async throws -> [QuestionDataModel]` and have that naming in UC and presentation layer as it tells you more about what you're doing. 
I would expect that `startQuizSession` triggers something and emits a new value asynchronously on a publisher somewhere else - i wouldn't expect that `start session` returns quiz questions and answers.